### PR TITLE
MB-46189: Azure - Added ability to select existing vnet

### PR DESCRIPTION
**Pull request by malscent**

### CODE REVIEW

File: azure/CouchbaseServerAndSyncGateway-Application/test/test_parameters/testExistingVnet.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
+{
+    "serverNodeCount": {
+      "value": 3
+    },
+    "serverDiskSize": {
+      "value": 30
+    },
+    "serverVersion": {
+      "value": "6.6.1"
```
Comment (patrick-stephens) [inline azure/CouchbaseServerAndSyncGateway-Application/test/test_parameters/testExistingVnet.json:9]: 6.6.2?

File: azure/CouchbaseServerAndSyncGateway-Application/test/test_parameters/testExistingVnet.json [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    },
+    "syncGatewayNodeCount": {
+      "value": 2
+    },
+    "syncGatewayVersion": {
+      "value": "2.8.1"
```
Comment (patrick-stephens) [inline azure/CouchbaseServerAndSyncGateway-Application/test/test_parameters/testExistingVnet.json:15]: Isn't this embargoed?

**Review by patrick-stephens — COMMENTED** [review]
I presume the test stuff is ok?

File: azure/CouchbaseServerAndSyncGateway-Application/test/test_parameters/testExistingVnet.json (hunk 1 above)
Comment (malscent) [reply azure/CouchbaseServerAndSyncGateway-Application/test/test_parameters/testExistingVnet.json:9]: oops

File: azure/CouchbaseServerAndSyncGateway-Application/test/test_parameters/testExistingVnet.json (hunk 2 above)
Comment (malscent) [reply azure/CouchbaseServerAndSyncGateway-Application/test/test_parameters/testExistingVnet.json:15]: yes

File: azure/CouchbaseServerAndSyncGateway-Application/mainTemplate.json [hunk 3]
```diff
@@ -279,24 +300,23 @@
...
+      "condition": "[equals(parameters('vnet').newOrExisting, 'new')]",
       "apiVersion": "2019-04-01",
       "location": "[parameters('location')]",
       "dependsOn": [
         "[resourceId('Microsoft.Network/networkSecurityGroups/', couchbase.generateResourceName('networkSecurityGroups'))]"
       ],
       "properties": {
         "addressSpace": {
-          "addressPrefixes": [
-            "10.0.0.0/23"
```
Comment (patrick-stephens) [inline azure/CouchbaseServerAndSyncGateway-Application/mainTemplate.json:292]: Have you changed this default now and does it matter?

File: azure/CouchbaseServerAndSyncGateway-Application/mainTemplate.json [hunk 4]
```diff
@@ -279,24 +300,23 @@
...
-          ]
+          "addressPrefixes": "[parameters('vnet').addressPrefixes]"
         },
         "subnets": [
           {
-            "name": "[couchbase.generateResourceName('subnet')]",
+            "name": "[parameters('vnet').subnets.subnet1.name]",
             "properties": {
-              "addressPrefix": "10.0.0.0/24",
+              "addressPrefix": "[parameters('vnet').subnets.subnet1.addressPrefix]",
```
Comment (patrick-stephens) [inline azure/CouchbaseServerAndSyncGateway-Application/mainTemplate.json:319]: So it looks like your defaults are now different - is that affecting anything?

File: azure/CouchbaseServerAndSyncGateway-Application/test/test_parameters/testExistingVnet.json [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+          "subnet1": {
+            "name": "cb-test",
+            "addressPrefix": "10.32.0.0/24",
+            "startAddress": "10.32.0.4"
+          }
+        }
+      }
+    },
+    "license": {
+      "value": "byol_2019"
```
Comment (patrick-stephens) [inline azure/CouchbaseServerAndSyncGateway-Application/test/test_parameters/testExistingVnet.json:48]: lol - should this be 2021?

File: azure/CouchbaseServerAndSyncGateway-Application/mainTemplate.json (hunk 4 above)
Comment (malscent) [reply azure/CouchbaseServerAndSyncGateway-Application/mainTemplate.json:319]: Nope.. we're using the standard defaults from Azure now.  Probably more correct.

File: azure/CouchbaseServerAndSyncGateway-Application/test/test_parameters/testExistingVnet.json (hunk 5 above)
Comment (malscent) [reply azure/CouchbaseServerAndSyncGateway-Application/test/test_parameters/testExistingVnet.json:48]: nope.. the license is called byol_2019, so it maps to what the marketplace says it is.

**Pull request merged**